### DISCUSSION TITLE
GitHub Actions with Podman Improvements

### DISCUSSION THREAD

**Discussion by p5** [containers/podman#22675] [General]
Hi Red Hat Team! 

# Problem

GitHub Actions currently run on Ubuntu 22.04 runners (soon to be 24.04 later this year). **The Ubuntu repos have an extremely outdated version of Podman available. (Podman 3.0)**

With the recent release of the Bootc images, there is an expectation for users to extend images in their own CI environment. The incompatibilities with the older version of Podman building Fedora 40 images have caused users to face problems when building images.

# Questions

**Do you happen to have any plans internally of improving support of Podman in GitHub Actions?**

I have concerns that users will use Docker or will resort to installing Podman from unofficial and untrusted locations to solve the problems above.

For example, in the [CentOS/centos-bootc](https://github.com/CentOS/centos-bootc/) repository, they are getting around this issue by installing Podman from OpenSUSE's unstable devel repository.

# Proposed Solutions

1. A first-party setup-podman action, which installs Podman without depending on the ubuntu apt repository in the same way that there's a setup-node and setup-docker-buildx action
2. Some documentation on your website with whichever way you recommend installing a newer podman version

Thank you for your time! 

Robert

**Comment by Luap99** [comment]
Well who is doing the work to maintain this? The fact is there is nobody who volunteers to maintain and fully support a repository to ship the latest versions on ubuntu/debian, https://github.com/containers/podman/discussions/17362

**Reply by p5** [reply]
Apologies if the description made it sound like I was pointing the finger.  That was not the intention at all.

The purpose of this discussion is to try to convey the problems around running Podman in GitHub Actions, and what we can do together (as a community and as maintainers) to help improve the experience.  This starts with knowing whether there are any existing plans or work in the pipeline which could help.

As it stands right now, Podman in GitHub Actions is quite frankly not a good experience.  
Given the recent announcements with Podman Desktop's integration with Bootc Image Builder, it sounds like Red Hat is pushing for Podman to be the go-to foundation for the new RHEL AI and Bootc services.  A large portion of that is the automation that sits behind the scenes, in the form of CI/CD pipelines.  It seems a bit strange that Red Hat wouldn't want to make (or help support efforts to make) the experience with Podman and RHEL AI/Bootc across all CI providers as good as possible.

If the recommendation is to simply not use Podman in GitHub Actions, then it would be beneficial to document that somewhere - such as the `redhat-actions` repositories - to avoid others going too far down the rabbit hole.

**Reply by Luap99** [reply]
> The purpose of this discussion is to try to convey the problems around running Podman in GitHub Actions, and what we can do together (as a community and as maintainers) to help improve the experience. This starts with knowing whether there are any existing plans or work in the pipeline which could help.

There are no plans to my knowledge. Anyone can package podman for debian/ubuntu and/or build a github action for it. I (personally) as upstream maintainer however have little to no interest in doing that because this is not easy and takes a lot of time and effort. I much rather fix real bugs and work on features in the actual code base.

And in general I cannot speak to Red Hat's company strategy and how using podman in github actions would fit into it. The podman team does not maintain `redhat-actions` AFAIK so maybe you could ask there as well.

**Comment by cgwalters** [comment]
Bigger picture, this is somewhat shortly going to be "solved" by https://github.blog/changelog/2024-05-14-github-hosted-runners-public-beta-of-ubuntu-24-04-is-now-available/

Of course, that version will eventually get out of date too, and we're likely to still add new interesting features to podman on the build and runtime side, but...in the short term it will help a lot.

**Reply by Luap99** [reply]
That version is already out of date, sure a lot better then 3.0 but still not 5.X
And as some users will notice one way or another debian builds are patched in ways that remove certain features so don't trust the version number to much.
